cognite/client/data_classes/assets.py

```diff
@@ -338,6 +339,22 @@ def labels(self) -> _LabelAssetUpdate:
     def geo_location(self) -> _PrimitiveAssetUpdate:
         return AssetUpdate._PrimitiveAssetUpdate(self, "geoLocation")
 
+    @classmethod
+    def _get_update_properties(cls) -> list[PropertySpec]:
+        return [
+            # External ID is nullable, but is used in the upsert logic and thus cannot be nulled out.
+            PropertySpec("external_id", is_nullable=False),
+            PropertySpec("name", is_nullable=False),
+            PropertySpec("description"),
+            PropertySpec("data_set_id"),
+            PropertySpec("metadata", is_list=True),
```

Comment:
The `is_list` name feels odd. What about `is_object` or `is_container`?

Reply:
I think `is_object` is misleading. Not a strong preference, although I lean towards staying `is_list`. @erlendvollset Thought?

Reply:
I think `is_primitive` would be the opposite of `is_list`? In that case I will go with `is_container`